packages/checkout-storefront/src/sections/CheckoutForm/useCheckoutSubmit.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  useCheckoutUpdateState,
+  useCheckoutUpdateStateActions,
+} from "@/checkout-storefront/state/updateStateStore";
+import {
+  useCheckoutValidationActions,
+  useCheckoutValidationState,
+} from "@/checkout-storefront/state/checkoutValidationStateStore";
+import { useCallback, useEffect, useState } from "react";
+import { useCheckoutFinalize } from "@/checkout-storefront/sections/CheckoutForm/useCheckoutFinalize";
+
+export const useCheckoutSubmit = () => {
+  const { validateAllForms } = useCheckoutValidationActions();
+  const { validating, validationState } = useCheckoutValidationState();
+  const { updateState, loadingCheckout } = useCheckoutUpdateState();
+  const { setShouldRegisterUser } = useCheckoutUpdateStateActions();
+  const { checkoutFinalize, finalizing } = useCheckoutFinalize();
+
+  const [submitInProgress, setSubmitInProgress] = useState(false);
+
+  const submitInitialize = useCallback(() => {
+    setSubmitInProgress(true);
+    setShouldRegisterUser(true);
+    validateAllForms();
+  }, [setShouldRegisterUser, validateAllForms]);
+
+  const updateStateValues = Object.values(updateState);
+
+  const anyRequestsInProgress =
+    updateStateValues.some((status) => status === "loading") || loadingCheckout;
+
+  const finishedApiChangesWithNoError =
+    !anyRequestsInProgress && updateStateValues.every((status) => status === "success");
+
+  const allFormsValid =
```

Comment:
Maybe we should have hooks for these utilities?

```
useAreAnyRequestsInProgres()
useFinishedApiChangesWithNoError()
useAreAllFormsValid()
```

etc.

Reply:
I don't think that there is a good reason to extract those to hooks. They're basically one line conditions so I'd say a hook for each would be an overkill. If it gets more crowded here I'll extract them to one utility hook tho